crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -32,6 +32,73 @@ def role_constraints
     end
   end
 
+  def shutdown_openstack_at_nodes
+    db_nodes = []
+    NodeObject.find("state:crowbar_upgrade").each do |node|
```

Comment:
Do we need to make sure here, to only run node nodes that have  ["crowbar_wall"]["crowbar_upgrade"] and non of the other upgrade related flags? E.g. what can go wrong if we reapply the recipe to a node that already had "crowbar_openstack_shutdown" set?